### DISCUSSION TITLE
New ideas for the project

### DISCUSSION THREAD

**Discussion by CypherRazzor** [cyberjunky/3commas-cyber-bots#171] [Ideas]
Hi Cyberjunky,

i use this scriptsuite for one day. Im very happy with what you have built, it makes my life as a botuser easier. I have some ideas, which would be helpful for me. I dont know wether they are relevant for other people.

- Automate signals from discord channels instead of telegram channels
- Close smart trades, when they are waiting for longer than x hours or days
- Close running smart trades at market price, when the trade is not closed after x hours or days
- TrailingStopLoss and TakeProfit for smart trades
- Automatic Naming of Bots
I name my bots for example: M: <amount of money used ex: 111> - <signal name ex: TV RSI> - <used coins ex: USDT_ALL> - <botname> - <exchangename>. When i use the coumpounder it changes the amount of money and would like to get the name changed.

Best regards

Cypher

**Comment by amargedon** [comment]
Hi! Good to read you are using the scripts and have some thoughts. Not sure if reading from Discord is something to be implemented soon actually. The ideas regarding smarttrades are interesting, but I think 3C already has the option for trailing SL and TP there. What is it you are currently missing there? Is that the configuration for the percentage of trailing?

**Comment by mlesnews** [comment]
3c dca bots have "Close deal after timeout | 'x' hours" (I use 29), and have the "[x] convert into smart trade."  I've not tested if will carry the timeout over from the dca bot to smart trade.
